src/NUnitFramework/framework/Attributes/PlatformAttribute.cs

```diff
@@ -56,12 +56,22 @@ public class PlatformAttribute : IncludeExcludeAttribute, IApplyToTest
         /// <param name="test">The test to modify</param>
         public void ApplyToTest(Test test)
         {
-            if (test.RunState != RunState.NotRunnable && 
-                test.RunState != RunState.Ignored && 
-                !platformHelper.IsPlatformSupported(this))
+            if (test.RunState != RunState.NotRunnable &&
+                test.RunState != RunState.Ignored)
             {
-                test.RunState = RunState.Skipped;
-                test.Properties.Add(PropertyNames.SkipReason, platformHelper.Reason);
+                try
+                {
+                    if (!platformHelper.IsPlatformSupported(this))
+                    {
+                        test.RunState = RunState.Skipped;
+                        test.Properties.Add(PropertyNames.SkipReason, platformHelper.Reason);
+                    }
+                }
+                catch (InvalidPlatformException ex)
```

Comment:
Since we went this route, it would make the code cleaner if `isPlatformSupported = platformHelper.IsPlatformSupported(this)` was the only thing inside the `try` block so that people understand the API usage by looking at the call site.